.github/workflows/deploy_staging.yml

```diff
@@ -0,0 +1,48 @@
+name: Deploy staging app to Firebase App Distribution
+on:
+  push:
+    branches:
+      - develop
+
+jobs:
+  deploy_android:
+    name: Deploy Android app to Firebase App Distribution
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    defaults:
+      run:
+        working-directory: ./sample
+    steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+      - name: Checkout source code
+        uses: actions/checkout@v2.3.2
+
+      - name: Cache Gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches/modules-*
+            ~/.gradle/caches/jars-*
+            ~/.gradle/caches/build-cache-*
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Decode buildKonfig properties
+        run: echo ${{ secrets.BUILD_KONFIG_PROPERTIES }} | base64 --decode > buildKonfig.properties
+
+      - name: Build staging APK
+        run: ./gradlew assembleStagingDebug -PversionCode=$GITHUB_RUN_NUMBER
+
+      - name: Deploy staging to Firebase
+        uses: wzieba/Firebase-Distribution-Github-Action@v1.7.0
+        with:
+          appId: ${{secrets.FIREBASE_APP_ID_STAGING}}
```

Comment:
This should be re-config in each env too https://github.com/nimblehq/kmm-templates/settings/environments